### DISCUSSION TITLE
When using `stdlib.OpenDBFromPool`, should the database be closed in the same function?

### DISCUSSION THREAD

**Discussion by meesvandongen** [jackc/pgx#2091] [Q&A]
The test (https://github.com/jackc/pgx/commit/639691c0abfb80dd7d7989904b6d9e58d4af9f43) within the `pgx` package does suggest that it should be closed.

The standard package says the following
> It is rare to Close a [DB], as the [DB] handle is meant to be long-lived and shared between many goroutines.

So I am not sure if it is valid to not close the db; and if it is, what the performance implications are.

**Comment by jackc** [comment]
Typically, your process will have a single `sql.DB` and/or `pgxpool.Pool` for it's entire lifetime. From that kind of Go process point of view it is fine to never clean them up.

However, it _may_ be better for your PostgreSQL server to disconnect via the PostgreSQL protocol rather than relying on whatever the OS does to TCP connections when the process ends.

As far as the test goes, we do want to clean up before the end of the process as there may be many other tests to run and we don't want those connection slots on the server to be consumed.
